cmd/immuclient/immuclienttest/helper.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2019-2020 vChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package immuclienttest
+
+import (
+	"bytes"
+	"io"
+	"log"
+	"os"
+	"sync"
+
+	"github.com/codenotary/immudb/cmd/helper"
+	"github.com/codenotary/immudb/cmd/immuclient/immuc"
+	"github.com/codenotary/immudb/pkg/client"
+	"github.com/codenotary/immudb/pkg/server/servertest"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+)
+
+type homedirServiceMock struct {
+	client.HomedirService
+	token []byte
+}
+
+func (h *homedirServiceMock) FileExistsInUserHomeDir(pathToFile string) (bool, error) {
+	return true, nil
+}
+
+func (h *homedirServiceMock) WriteFileToUserHomeDir(content []byte, pathToFile string) error {
+	h.token = content
+	return nil
+}
+
+func (h *homedirServiceMock) DeleteFileFromUserHomeDir(pathToFile string) error {
+	return nil
+}
+
+func (h *homedirServiceMock) ReadFileFromUserHomeDir(pathToFile string) (string, error) {
+	return string(h.token), nil
+}
+
+func NewClient(pr helper.PasswordReader, dialer servertest.BuffDialer, hds client.HomedirService) immuc.Client {
```

Comment:
This is confusing to me:
there is NewClient method that create new client from a PasswordReader, BuffDialer and HomedirService
then we have Login that do the same stuff of NewClient + login but it returns also HomedirService

We should have a struct with following exposed dependencies:
- client
- clientOptions
- tokenfile
- homeDirService(if you need it in a second time)

end on the receiver we could expose following methods:
- connect
- login

Feel free to improve or rearrange if you like